tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -62,26 +64,70 @@ check_for_system_columns(Bitmapset *attrs_used)
 	}
 }
 
+/*
+ * Scratch space for mapping out the decompressed columns.
+ */
+typedef struct
+{
+	/*
+	 * decompression_map maps targetlist entries of the compressed scan to tuple
+	 * attribute number of the uncompressed chunk. Negative values are special
+	 * columns in the compressed scan that do not have a representation in the
+	 * uncompressed chunk, but are still used for decompression.
+	 */
+	List *decompression_map;
+
+	/*
+	 * This Int list is parallel to the compressed scan targetlist, just like
+	 * the above one. The value is true if a given targetlist entry is a
+	 * segmentby column, false otherwise. Has the same length as the above list.
+	 * We have to use the parallel lists and not a list of structs, because the
+	 * Plans have to be copyable by the Postgres _copy functions, and we can't
+	 * do that for a custom struct.
+	 */
+	List *is_segmentby_column;
+
+	/*
+	 * Same structure as above, says whether we support bulk decompression for this
+	 * column.
+	 */
+	List *bulk_decompression_column;
+
+	/*
+	 * If we produce at least some columns that support bulk decompression.
+	 */
+	bool have_bulk_decompression_columns;
+
+	/*
+	 * Maps the uncompressed chunk attno to the respective column compression
+	 * info. This lives only during planning so that we can understand on which
+	 * columns we can apply vectorized quals.
+	 */
+	DecompressChunkColumnCompression *uncompressed_chunk_attno_to_compression_info;
+} DecompressionMapContext;
```

Comment:
It was always confusing to see these things in the Path where they are actually not valid, so I put them into a separate struct.